prometheus/handler.go

```diff
@@ -71,6 +85,13 @@ func (h *Handler) timeout() time.Duration {
 
 // ServeHTTP satsifies the http.Handler interface.
 func (h *Handler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	switch req.Method {
```

Comment:
Yep unrelated to trimming, but related to having defaults that do the right thing.